packages/server/lib/browsers/electron.ts

```diff
@@ -449,85 +450,87 @@ module.exports = {
   },
 
-  open (browser, url, options = {}, automation) {
+  async open (browser, url, options, automation) {
     const { projectRoot, isTextTerminal } = options
 
     debug('open %o', { browser, url })
 
-    return savedState.create(projectRoot, isTextTerminal)
-    .then((state) => {
-      return state.get()
-    }).then((state) => {
-      debug('received saved state %o', state)
+    const State = await savedState.create(projectRoot, isTextTerminal)
+    const state = await State.get()
 
-      // get our electron default options
-      // TODO: this is bad, don't mutate the options object
-      options = this._defaultOptions(projectRoot, state, options, automation)
+    debug('received saved state %o', state)
 
-      // get the GUI window defaults now
-      options = Windows.defaults(options)
+    // get our electron default options
+    // TODO: this is bad, don't mutate the options object
+    options = this._defaultOptions(projectRoot, state, options, automation)
 
-      debug('browser window options %o', _.omitBy(options, _.isFunction))
+    // get the GUI window defaults now
+    options = Windows.defaults(options)
 
-      const defaultLaunchOptions = utils.getDefaultLaunchOptions({
-        preferences: options,
-      })
+    debug('browser window options %o', _.omitBy(options, _.isFunction))
 
-      return utils.executeBeforeBrowserLaunch(browser, defaultLaunchOptions, options)
-    }).then((launchOptions) => {
-      const { preferences } = launchOptions
+    const defaultLaunchOptions = utils.getDefaultLaunchOptions({
+      preferences: options,
+    })
 
-      debug('launching browser window to url: %s', url)
+    const launchOptions = await utils.executeBeforeBrowserLaunch(browser, defaultLaunchOptions, options)
 
-      return this._render(url, automation, preferences, {
-        projectRoot: options.projectRoot,
-        isTextTerminal: options.isTextTerminal,
-      })
-      .then(async (win) => {
-        await _installExtensions(win, launchOptions.extensions, options)
+    const { preferences } = launchOptions
+
+    debug('launching browser window to url: %s', url)
 
-        // cause the webview to receive focus so that
-        // native browser focus + blur events fire correctly
-        // https://github.com/cypress-io/cypress/issues/1939
-        tryToCall(win, 'focusOnWebView')
+    const win = await this._render(url, automation, preferences, {
+      projectRoot: options.projectRoot,
+      isTextTerminal: options.isTextTerminal,
+    })
 
-        const events = new EE
+    await _installExtensions(win, launchOptions.extensions, options)
 
-        win.once('closed', () => {
-          debug('closed event fired')
+    // cause the webview to receive focus so that
+    // native browser focus + blur events fire correctly
+    // https://github.com/cypress-io/cypress/issues/1939
+    tryToCall(win, 'focusOnWebView')
 
-          Windows.removeAllExtensions(win)
+    const events = new EE
```

Comment:
```suggestion
    const events = new EE()
```

Trivial but I find `new` with `()` is so weird to look at :shrug: nitpicking, not a biggie